src/Product.php

```diff
@@ -228,4 +228,15 @@ public static function display_store_stock_status_block() {
     ]);
   }
 
+  /**
+   * Checks for excluded product categories.
+   *
+   * @return bool
+   *   Wether the product is assigned to any excluded category.
+   */
+  public static function isTermExcluded() {
+    $product_categories = wp_list_pluck(get_the_terms(get_the_id(), 'product_cat'), 'name');
```

Comment:
`get_the_terms()` can return `Wp_Error`, please check for it and also `FALSE` otherwise it could couse a critical error with `array_intersect`.

Reply:
```suggestion
    $product_categories = wp_list_pluck(get_the_terms(get_the_id(), 'product_cat'), 'name');
```

According to the documentation, get_the_terms returns a `WP_Term` object on success, but I cannot find product_cat among them, are you sure that `wp_list_pluck()` is giving you what you need? 

https://developer.wordpress.org/reference/classes/wp_term/#user-contributed-notes